### PR TITLE
feat(ui): export modal for all the entities

### PR DESCRIPTION
**Pull request by chirag-madlani**
### Describe your changes:

[#10021](https://github.com/open-metadata/OpenMetadata/issues/10021)

- Worked on centralizing export functionality
- Integrated export functionality in teams page

https://github.com/open-metadata/OpenMetadata/assets/71748675/03b0f56c-8f00-4e56-ab73-1cc58710d94c

#
### Type of change:

- [ ] Bug fix
- [ ] Improvement
- [ ] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [ ] My PR title is `Fixes <issue-number>: <short explanation>`
- [ ] I have commented on my code, particularly in hard-to-understand areas. 
- [ ] For JSON Schema changes: I updated the migration scripts or explained why it is not needed.

### CODE REVIEW

**Review by Sachin-chaurasiya — COMMENTED** [review]
Left a few comments, thanks

**Comment by ShaileshParmar11** [comment]
Hello, @Sachin-chaurasiya comments have been addressed!

**Comment by Sachin-chaurasiya** [comment]
> Hello, @Sachin-chaurasiya comments have been addressed!

Can we change the component file name as well?

File: openmetadata-ui/src/main/resources/ui/src/pages/teams/TeamsPage.tsx [hunk 1]
```diff
@@ -483,8 +483,8 @@ const TeamsPage = () => {
     history.push(getSettingPath(getTeamsWithFqnPath(TeamType.Organization)));
   };
 
-  const handleShowDeletedTeam = (checked: boolean) => {
-    setShowDeletedTeam(checked);
+  const handleShowDeletedTeam = () => {
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/pages/teams/TeamsPage.tsx:486]:
```suggestion
  const toggleShowDeletedTeam = () => {
```

File: openmetadata-ui/src/main/resources/ui/src/components/Entity/EntityExportModalProvider/EntityExportModalProvider.component.tsx [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+    URL.revokeObjectURL(element.href);
+    document.body.removeChild(element);
+  };
+
+  const handleExport = async ({ fileName }: { fileName: string }) => {
+    if (exportData === null) {
+      Promise.reject(t('message.api-must-provide-for-export'));
+
+      return;
+    }
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/components/Entity/EntityExportModalProvider/EntityExportModalProvider.component.tsx:70]: We have check for exportData before showing modal so don't think we have a case where the function is being called with exportData being null.
We can have then condition but no need to throw any error i belive

**Review by chirag-madlani — COMMENTED** [review]
Looks good

**Review by shahsank3t — APPROVED** [review]

**Pull request merged**